crypto_analytics/collection/data_source/crypto_compare.py

```diff
@@ -1,28 +1,31 @@
 import pandas as pd
 import json
 import requests
+from typing import Dict, Union
 
-from crypto_analytics.collection.data_source import FinancialDataSource
-from crypto_analytics.types import Interval
+from crypto_analytics.collection.data_source import OHLCDataSource
+from crypto_analytics.types  import Interval
 
-class CryptoCompare(FinancialDataSource):
+class CryptoCompareOHLC(OHLCDataSource):
+    endpoints = {
+        Interval.MINUTE: 'data/histominute',
+        Interval.HOURLY: 'data/histohour',
+        Interval.DAILY: 'data/histoday',
+    }
+
+    def __init__(self, interval: Interval, fsym: str, tsym: str, limit: int):
+        self.__prevalidate(interval, fsym, tsym, limit)
 
-    def __init__(self, endpoint: str, fsym: str, tsym: str, limit: int):
```

Comment:
I think the endpoint shouldn't be a parameter for our data sources just because different endpoints could return very different data and our data source classes usually make assumptions about what kind of data gets returned. If we want to support more crypto compare endpoints I think we can just make a new class in this file. This is also why I decided to rename `CryptoCompare` to `CryptoCompareOHLC` so that we have other classes for other CryptoCompare endpoint types.

Reply:
Are you saying we should make a new class for each endpoint or a new class that will hold all the endpoints and handle getting the data into a unified format?

Reply:
I think we should make a new class for each category of endpoints. For example `CryptoCompareOHLC` actually has three endpoints for daily, hourly and minute OHLC data but they all have the same OHLC columns so we can group these into one class. If we have an `endpoint` parameter, then the user might end up using an endpoint that doesn't have OHLC data which would violate assumptions we are making to conform to `OHLCDataSource`. So I prefer having the class handle which endpoint gets called rather than the user and to have different classes handle different types of data.

Reply:
Ok that sounds good.